End of history notification

hfakkiz opened the pull request:
This is final commit in this history line. The commit, and the old history will be accessible on branch zephyr-v4.0.0-snapshot and via tag zephyr-v4.0.0-snapshot.
The issue has been described in details here:
https://github.com/zephyrproject-rtos/zephyr/issues/85205

Briefly: Inner "hal_adi" and "zephyrproject-rtos/hal_adi" repositories have different histories. This causes a sychronization issue. To solve this, we have decide to reset the history by overwriting main branch of "zephyrproject-rtos/hal_adi" with inner "hal_adi" repository's history.

hfakkiz committed:
End of history notification

This is final commit in this history line. The commit, and the old
history will be accessible on branch zephyr-v4.0.0-snapshot and via tag
zephyr-v4.0.0-snapshot.
The issue has been described in details here:
https://github.com/zephyrproject-rtos/zephyr/issues/85205

Briefly: Inner "hal_adi" and "zephyrproject-rtos/hal_adi" repositories
have different histories. This causes a sychronization issue. To solve
this, we have decide to reset the history by overwriting main branch of
"zephyrproject-rtos/hal_adi" with inner "hal_adi" repository's history.

Signed-off-by: Furkan Akkiz <hasanfurkan.akkiz@analog.com>

hfakkiz commented:
@MaureenHelm Please take a look.

MaureenHelm reviewed (approved):
@ttmut @ozersa Please review

ozersa reviewed (changes requested):
I think this is not a good approach. We shall keep history and add feature on top of it instead removing past.

There is not a big sync problem that can not be resolved.

Simple solution can be.

1- analogdevicesinc/hal_adi/main is behind the zephyr/hal_adi/main there are ~8 commits here: These branches shall be synced. See PR: https://github.com/analogdevicesinc/hal_adi/pull/8 
2- Then analogdevices/hal_adi/develop branch shall be rebased onto main branch.
3- develop and main branch delta can be send to zephyr/hal_adi either commit by commit (I preferred that) or a bulk.

hfakkiz commented:
> I think this is not a good approach. We shall keep history and add feature on top of it instead removing past.
> 
> There is not a big sync problem that can not be resolved.
> 
> Simple solution can be.
> 
> 1- analogdevicesinc/hal_adi/main is behind the zephyr/hal_adi/main there are ~8 commits here: These branches shall be synced. See PR: [analogdevicesinc/hal_adi#8](https://github.com/analogdevicesinc/hal_adi/pull/8) 2- Then analogdevices/hal_adi/develop branch shall be rebased onto main branch. 3- develop and main branch delta can be send to zephyr/hal_adi either commit by commit (I preferred that) or a bulk.

The source of "hal_adi" repositories is [MSDK](https://github.com/analogdevicesinc/msdk) repo and currently [analogdevicesinc/hal_adi](https://github.com/analogdevicesinc/hal_adi/tree/develop) repo gets each commit of MSDK repo. But we added commits to [zephyrproject-rtos/hal_adi](https://github.com/zephyrproject-rtos/hal_Adi) repo manually for developed features. This made history of zephyrproject-rtos/hal_adi different from MSDK and analogdevicesinc/hal_adi repos.

We use [analogdevicesinc/hal_adi/develop](https://github.com/analogdevicesinc/hal_adi/tree/develop) branch internally but it has problems too. At the beginning, MSDK developments added into this branch as a bulk and now it gets each MSDK changes to analogdevicesinc/hal_adi without a development related commit description and author information. These problems cause traceability issue. To improve traceability, we created [analogdevicesinc/hal_adi/new_hal_adi_history](https://github.com/analogdevicesinc/hal_adi/tree/new_hal_adi_history) branch and made it has same commit history with MSDK repo. Also, we updated workflow and further MSDK changes will come with same commit description and author information. Now, we want to move forward with this structure for internal and Zephyr "hal_adi" repositories.
